pydantic_settings/sources.py

```diff
@@ -144,6 +144,61 @@ def _extract_field_info(self, field: FieldInfo, field_name: str) -> List[Tuple[s
 
         return field_info
 
+    def _replace_field_names_case_insensitively(self, field: FieldInfo, field_values: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Replace field names in values dict by looking in models fields insensitively.
+
+        By having the following models:
+
+            class SubSubSub(BaseModel):
+                VaL3: str
+
+            class SubSub(BaseModel):
+                Val2: str
+                SUB_sub_SuB: SubSubSub
+
+            class Sub(BaseModel):
+                VAL1: str
+                SUB_sub: SubSub
+
+            class Settings(BaseSettings):
+                nested: Sub
+
+                model_config = ConfigDict(env_nested_delimiter='__')
+
+        Then:
+            _replace_field_names_case_insensitively(
+                field,
+                {"val1": "v1", "sub_SUB": {"VAL2": "v2", "sub_SUB_sUb": {"vAl3": "v3"}}}
+            )
+            Returns {'VAL1': 'v1', 'SUB_sub': {'Val2': 'v2', 'SUB_sub_SuB': {'VaL3': 'v3'}}}
+        """
+        values: Dict[str, Any] = {}
+
+        for name, value in field_values.items():
+            sub_model_field: Optional[FieldInfo] = None
+
+            if not field.annotation:
+                values[name] = value
+                continue
```

Comment:
are you sure this is necessary? It looks like where this is called `field.annotations` is always a `BaseModel` - also the next line assumes `field.annotation.model_fields` always exists.

Reply:
It's here only to make mypy happy
```
Item "None" of "Optional[Type[Any]]" has no attribute "model_fields"
```

Reply:
Done!